### PR TITLE
update(opaque-tokens-in-angular-2): editorial improvements

### PR DESCRIPTION
**Pull request by ThomasBurleson**

### CODE REVIEW

**Comment by ThomasBurleson** [comment]
Attn: @PascalPrecht

File: _posts/2016-05-23-opaque-tokens-in-angular-2.md [hunk 1]
```diff
@@ -107,13 +107,15 @@ class MyComponent {
...
 {% endhighlight %}
 
+Note: that above we used `@Inject(featureEnabledToken) private featureEnabled` without any typing information; e.g. `private featureEnabled:boolean`.
+
 Okay awesome, we can use strings and types as tokens to inject dependencies in our application. Unfortunately, using string tokens like this, opens up potential for naming collisions.
 
 ## The problem with string tokens
 
-Let's get back to our `config` string token for a second. "config" is a pretty general name, so we probably could've done better naming this thing in the first place. However, even if we come up with a more distinguishable name, there's is literally no guarantee that someone else will use the same string as a token.
+Let's get back to our `config` string token for a second. "config" is a pretty general name, so we probably could've done better naming this thing in the first place. However, even if we come up with a more distinguishable name, there is easily possible that someone else will use the same string as a token. Providers are flattened, meaning that, if there are multiple providers with the same token, the last one wins.
```
Comment (ThomasBurleson) [inline _posts/2016-05-23-opaque-tokens-in-angular-2.md]: it is easily possible.

File: _posts/2016-05-23-opaque-tokens-in-angular-2.md [hunk 2]
```diff
@@ -44,15 +44,15 @@ class MyComponent {
...
+Since the token matches the dependency instance-type and the provider strategy is `useClass`, we can actually also the **shorthand** version, as:
 
 {% highlight js %}
 {% raw %}
 providers: [DataService]
 {% endraw %}
 {% endhighlight %}
 
-That's cool, as long as we have classes (or types) to represent the things we want to work with. However, sometimes we need to create other objects that don't necessarily need to be put in a class representation. We could for example have a configuration object that we want to inject into our application. This configuration object can be a simple object literal, there's no type involved.
+Angular2 has many shorthand versions; and ^ is just one of those. Now this is cool, as long as we have classes (or types) to represent the things we want to work with. However, sometimes we need to create other objects that don't necessarily need to be put in a class representation. We could for example have a configuration object that we want to inject into our application. This configuration object can be a simple object literal where there is no type involved.
```
Comment (ThomasBurleson) [inline _posts/2016-05-23-opaque-tokens-in-angular-2.md]: and above is just one examples of those.

File: _posts/2016-05-23-opaque-tokens-in-angular-2.md [hunk 3]
```diff
@@ -21,13 +21,13 @@ author: pascal_precht
...
 
-Let's first recap what the differents between a string token and a type token is, before we jump into the actual problem we want to solve.
+Before we jump into the actual problem we want to solve, let's first recap the differences between a string token and a type token.
 
 ## String Tokens vs Type Tokens
 
-In order to associate a dependency creation with a token that we use throughout our application, we have to setup providers. A couple weeks ago we've written about how [providers can be created using Map literals](/angular/2016/05/13/angular-2-providers-using-map-literals.html), if you haven't read this one yet we recommend to check it out, as this article pretty much builds up on that one.
+Angular 2 uses dependency factories to create singleton instances that will be used in the runtime DI processes. In order to associate a token with a dependency factory, we have to setup providers. A couple weeks ago we blogged how [providers can be created using Map literals](/angular/2016/05/13/angular-2-providers-using-map-literals.html), if you haven't read this one yet we recommend to check it out, as this article pretty much builds up on that one.
```
Comment (ThomasBurleson) [inline _posts/2016-05-23-opaque-tokens-in-angular-2.md]: For DI, Angular 2 uses factories to create singleton instances

File: _posts/2016-05-23-opaque-tokens-in-angular-2.md [hunk 4]
```diff
@@ -44,15 +44,15 @@ class MyComponent {
 {% endraw %}
 {% endhighlight %}
 
-Or, in this particular case, we can also use the shorthand version, as the token matches the dependency instance type and the provider strategy is `useClass`:
+Since the token matches the dependency instance-type and the provider strategy is `useClass`, we can actually also the **shorthand** version, as:
```
Comment (PascalPrecht) [inline _posts/2016-05-23-opaque-tokens-in-angular-2.md]: there's a missing "use"

File: _posts/2016-05-23-opaque-tokens-in-angular-2.md (hunk 2 above)
Comment (PascalPrecht) [inline _posts/2016-05-23-opaque-tokens-in-angular-2.md]: That's new to me. Actually there's only one shorthand, which is the one showed. Everything that doesn't match that rule needs to go with the more verbose syntax.

**Comment by PascalPrecht** [comment]
Thanks @ThomasBurleson ! I left some comments. Let's get those clarified, then we can merge this right away!

File: _posts/2016-05-23-opaque-tokens-in-angular-2.md [hunk 5]
```diff
@@ -156,7 +158,7 @@ This will pretty much break our third-party library, because now, the thing that
 
 ## Opaque Tokens to the resque
 
-Luckily, Angular got us covered. It comes with a type called `OpaqueToken` that basically allows us to create string based tokens without running into any collisions.
+Luckily, Angular got us covered. It comes with a type called `OpaqueToken` that basically allows us to create string-based tokens without running into any collisions.
```
Comment (ThomasBurleson) [inline _posts/2016-05-23-opaque-tokens-in-angular-2.md]: Luckily, Angular anticipated such scenarios.

File: _posts/2016-05-23-opaque-tokens-in-angular-2.md (hunk 3 above)
Comment (PascalPrecht) [inline _posts/2016-05-23-opaque-tokens-in-angular-2.md]: I think the term "factories" might confuse ppl here. Even though, DI normalizes provider recipes under the hood to "factories"

Comment (ThomasBurleson) [inline _posts/2016-05-23-opaque-tokens-in-angular-2.md]: The important aspect here is that "DI injects singleton instances which are created by provider-registered factories."

File: _posts/2016-05-23-opaque-tokens-in-angular-2.md (hunk 2 above)
Comment (ThomasBurleson) [inline _posts/2016-05-23-opaque-tokens-in-angular-2.md]: I meant that Angular 2 has many shorthand versions:  1 for providers, some for annotations, etc.

**Comment by PascalPrecht** [comment]
LGTM merged!

**Pull request merged**